scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1555,7 +1527,8 @@ else if (connection.isResumptionRequired()){
 
 			// terminate the previous connection and add the new one to the store
 			Connection newConnection = new Connection(peerAddress, config.getAutoResumptionTimeoutMillis());
-			terminateConnection(connection, null, null);
+			connection.cancelPendingFlight();
+			connectionStore.remove(connection, false);
```

Comment:
This change is because in this case we don't want to remove session from sessionCache ?

Reply:
Yes.